cedar-dafny/validation/thm/model.dfy

```diff
@@ -46,11 +46,24 @@ module validation.thm.model {
     if (es[0].0 == k && (forall j | 0 < j < |es| :: es[j].0 != k)) then es[0].1 else LastOfKey(k,es[1..])
   }
 
+  lemma InterpretRecordPreservesAttrs(es: seq<(Attr, Expr)>, r: Request, s: EntityStore)
+    requires Evaluator(r,s).interpretRecord(es).Ok?
+    ensures forall i :: 0 <= i < |es| ==> es[i].0 in Evaluator(r,s).interpretRecord(es).value.Keys
+    ensures forall k | k in Evaluator(r,s).interpretRecord(es).value.Keys :: KeyExists(k,es)
+  {
+    if es != [] {
```

Comment:
Nit: maybe `|es| == 0` looks better?

Reply:
FWIW, I prefer John's approach:  compare directly to the empty value.